src/Microsoft.ML.LightGBM/Microsoft.ML.LightGBM.csproj

```diff
@@ -0,0 +1,21 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFramework>netstandard2.0</TargetFramework>
+    <IncludeInPackage>Microsoft.ML.LightGBM</IncludeInPackage>
+	<AllowUnsafeBlocks>true</AllowUnsafeBlocks>
+  </PropertyGroup>
+  
+  <ItemGroup>
+    <ProjectReference Include="..\Microsoft.ML.Core\Microsoft.ML.Core.csproj" />
+    <ProjectReference Include="..\Microsoft.ML.Data\Microsoft.ML.Data.csproj" />
+    <ProjectReference Include="..\Microsoft.ML.FastTree\Microsoft.ML.FastTree.csproj" />
```

Comment:
Why does LightGBM have a dependency on FastTree?

Reply:
As you might imagine, there were a number of existing facilities for FastTree predictors (both in this code, and elsewhere). So when Goulin and Taifeng were first coming up with this, they chose to actually return an actual FastTree predictor, rather than invent their own. The shared format between the two learners has many benefits, but it does result in this dependency unfortunately. A cleaner solution might have been to factor out the code so that it is outside the FastTree project, but that code is so awful (still 99% of it in its original MSR state) that possibly they found this a daunting prospect. (Certainly I would. :) )

---
In reply to: [198167265](https://github.com/dotnet/machinelearning/pull/392#discussion_r198167265) [](ancestors = 198167265)

Reply:
That is right, the predictor is shared between FastTree and LightGBM, refer to regressiontree.cs in FastTree project.

---
In reply to: [198213703](https://github.com/dotnet/machinelearning/pull/392#discussion_r198213703) [](ancestors = 198213703,198167265)